ros_ethercat_model/include/ros_ethercat_model/ros_ethercat.hpp

```diff
@@ -182,6 +182,54 @@ class RosEthercat : public hardware_interface::RobotHW
     }
   }
 
+  /// Update priority of provided thread
+  bool updateThreadPriority(boost::thread& a_thread)
+  {
+    int retcode;
+    int policy;
+    struct sched_param param;
+
+    pthread_t threadID = (pthread_t) a_thread.native_handle();
+
+    policy = SCHED_FIFO;
+    param.sched_priority = sched_get_priority_max(policy);
+
+    if ((retcode = pthread_setschedparam(threadID, policy, &param)) != 0)
+    {
+      ROS_ERROR("Error setting policy/priority of Ethercat hardware threads. Please restart the system");
+      return false;
+    }
+    return true;
+  }
+
+  /// Thread that calls EthercatHardware.update
+  void ethercat_update_thread(EthercatHardware * eh)
+  {
+    while (true)
+    {
+      {
```

Comment:
I think the extra curly braces mean there's some C++ stuff going on here that I don't understand, if you get a second could you explain it to me?

Reply:
Oh that was to contain the lock within that scope only (I think that's required?)

Reply:
ah ok, so it's so that the lock will automatically get destroyed as it goes out of scope a few lines later?

Reply:
yes!